### PR TITLE
LMPOP

### PR DESCRIPTION
**Pull request by Razi007**

### CODE REVIEW

**Review by sazzad16 — COMMENTED** [review]
From https://redis.io/commands/lmpop

> A two-element array with the first element being the name of the key from which elements were popped, and the second element is an array of elements.

In summary, this is a single list along with respective key which makes just a single map entry.

@Razi007 Why do want to return an entire `Map` with just one entry?

**Comment by Razi007** [comment]
> From https://redis.io/commands/lmpop
> 
> > A two-element array with the first element being the name of the key from which elements were popped, and the second element is an array of elements.
> 
> In summary, this is a single list along with respective key which makes just a single map entry.
> 
> @Razi007 Why do want to return an entire `Map` with just one entry?

@sazzad16 Got your point.I was assuming it might return multiple keys with respective elements.Changing it and pushing the code for the same.

**Comment by Razi007** [comment]
> > From https://redis.io/commands/lmpop
> > > A two-element array with the first element being the name of the key from which elements were popped, and the second element is an array of elements.
> > 
> > 
> > In summary, this is a single list along with respective key which makes just a single map entry.
> > @Razi007 Why do want to return an entire `Map` with just one entry?
> 
> @sazzad16 Got your point.I was assuming it might return multiple keys with respective elements.Changing it and pushing the code for the same.

File: src/main/java/redis/clients/jedis/resps/KeyedListElement.java [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
+import java.awt.*;
+
 /**
  * This class is used to represent a List element when it is returned with respective key name.
  */
 public class KeyedListElement {
 
   private final String key;
   private final String element;
+  private final List elementList;
```
Comment (Razi007) [inline src/main/java/redis/clients/jedis/resps/KeyedListElement.java:14]: @sazzad16 added a new list value type here.Please suggest can we proceed this way.

**Comment by Razi007** [comment]
closed by mistake

File: src/main/java/redis/clients/jedis/resps/KeyedListElement.java (hunk 1 above)
Comment (sazzad16) [reply src/main/java/redis/clients/jedis/resps/KeyedListElement.java:15]: @Razi007 I don't like how it has been handled later, like `this.elementList = new List();` and `this.element = "";`.

**Review by sazzad16 — COMMENTED** [review]
@Razi007 Your methods for binary variants have issues which are not apparent due to lack of tests. A personal suggestion, Implement String variants (without Pipeline) first. After this is done properly, then implement binary ones. And after that Pipeline variants.

File: src/main/java/redis/clients/jedis/resps/KeyedListValueElements.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+import java.util.List;
+
+
+public class KeyedListValueElements<K,V> {
+
+  private final K key;
+  private final List<V> elements;
+
```
Comment (Razi007) [inline src/main/java/redis/clients/jedis/resps/KeyedListValueElements.java:11]: @sazzad16 implemented a new class here.Using generic key values.

File: src/test/java/redis/clients/jedis/commands/jedis/ListCommandsTest.java [hunk 3]
```diff
@@ -875,4 +876,25 @@ public void blmove() {
     assertByteArrayListEquals(Collections.singletonList(b3), jedis.lrange(bbar, 0, -1));
     assertByteArrayListEquals(Arrays.asList(b1, b2), jedis.lrange(bfoo, 0, -1));
   }
+
+  @Test
+  public void lmpop(){
+    KeyedListValueElements keyedListValueElements;
```
Comment (Razi007) [inline src/test/java/redis/clients/jedis/commands/jedis/ListCommandsTest.java:882]: @sazzad16 added all test cases here.String args are working fine.Please check the approach

**Comment by sazzad16** [comment]
Closed by a08fa3e

**Pull request closed without merging**